real_estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,147 @@
+    )
+    # SQL constraint to ensure the price is strictly positive
+    _sql_constraints = [
+        (
+            'check_offer_price',
+            'CHECK(price > 0)',
+            "The Offer Price must be strictly positive"
+        )
+    ]
+
+    @api.depends('create_date', 'validity')
+    def _compute_date_deadline(self):
+        """Compute the offer's deadline date as the creation date + validity days.
+        
+        If the offer has no creation date (e.g., a draft record), the deadline is not set.
+        """
+        for offer in self:
+            if date.today():  # Ensure today's date is available
+                offer.date_deadline = date.today() + timedelta(days=offer.validity)
+            else:
+                offer.date_deadline = False
+
+    def _inverse_date_deadline(self):
+        """Update the validity period based on the difference between deadline and creation date.
+        
+        If there is no valid deadline or creation date, reset the validity to 7 days.
+        """
+        for offer in self:
+            if offer.date_deadline and offer.create_date:
+                offer.validity = (offer.date_deadline - offer.create_date.date()).days
+            else:
+                offer.validity = 7
+
+    def action_confirm(self):
+        """Accept an offer and update the property's state accordingly.
+        
+        - Ensures that only one offer can be accepted per property.
+        - Updates the property's state, buyer, and selling price.
+        """
+        for offer in self:
+            if offer.property_id.offer_ids.filtered(lambda o: o.status == 'accept'):
+                raise UserError("Only one offer can be accepted per property.")
+            offer.status = 'accept'
+            offer.property_id.state = 'offer_accepted'
+            offer.property_id.buyer_id = offer.partner_id
+            offer.property_id.selling_price = offer.price
+
+    def action_reject(self):
+        """Reject an offer by setting its status to 'reject'."""
+        self.status = 'reject'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Override the create method to enforce offer price validation.
+        
+        - Ensures that the offer price is higher than any existing offers.
+        - Updates the property's state to 'offer_received' upon a valid offer.
+        """
+        for vals in vals_list:
+            property = self.env['estate.property'].browse(vals.get('property_id'))
+            if property.best_price >= vals['price']:
+                raise UserError("Offer Price is lower than the existing ones")
+            property.state = 'offer_received'
+        return super().create(vals_list)
+    
+    @api.constrains('property_type_id','partner_id')
+    def _check_commercial_buyer(self):
+        for offer in self:
+            if(
+                offer.property_type_id==offer.env.ref('real_estate.property_type_commercial').id
```

Comment:
```suggestion
                offer.property_type_id == offer.env.ref('real_estate.property_type_commercial').id
```
styling check
does this even work?
you are comparing object with `int`